### PR TITLE
Bigquery refactor

### PR DESCRIPTION
**Pull request by tuvaforrest**
## Describe your changes
updated a couple things to support bigquery

## How has this been tested?
ran on redshift, bigquery, and snowflake

## Reviewer focus
Run on each if you like.  Let me know if you think the new macro for date_part is good or if we should cast the output natively or 

## Checklist before requesting a review
- [x]  I have recorded a Loom performing a self-review of my code
- [x]  My code follows style guidelines
- [x]  I have commented my code as necessary
- [x]  I have testing by code by running `dbt build` in the following warehouse(s):
    - [x]  Snowflake
    - [x]  Redshift
    - [x]  Bigquery
- [x]  I have implemented generic dbt tests to validate primary keys/uniqueness in my model
- [x]  I have updated dbt docs by running `dbt docs generate` and copying the appropriate files to the `docs/` path
- [x]  I have added at least one Github label to this PR

## (Optional) Gif of how this PR makes you feel
gif

(because the panda is big!)

## Loom link

Note, since I recorded the loom I made a few small change: I added Bigquery to the readme, updated the version number in the dbt_project.yml to reflect the next release, and removed trailing whitespace in the variables that may have been causing an issue in another package for a customer (though I CNR).

https://www.loom.com/share/2eafb2d345f24a72be293d6dbb976f37

## For reviewers:
[Best practices for reviewing pull requests.](https://www.notion.so/Pull-Request-Review-Best-Practices-b486354d76c04982889f2178fe777c1c)

### CODE REVIEW

**Review by aneiderhiser — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ For information on data models and to view the entire DAG check out our dbt [Doc
```
Comment (thutuva) [inline README.md:24]: The release version in the ReadMe needs to be updated.

(I couldn't make a comment there since my GitHub is only showing diffs)

**Review by thutuva — CHANGES_REQUESTED** [review]
Just need the package version updated in the ReadMe.

I also tested in BQ, redshift, and PMPM since my CI was not ready for this PR.

File: README.md (hunk 1 above)
Comment (tuvaforrest) [reply README.md:24]: updated

**Review by thutuva — APPROVED** [review]
All set, thanks!

**Pull request merged**